### PR TITLE
feat(performance): add feature flag for perf work

### PR DESCRIPTION
**Pull request by slohmes**
Create a new feature flag, "enterprise-perf", for enterprise team's
 performance improvement efforts. Off by default.

### CODE REVIEW

**Comment by slohmes** [comment]
Lmk if there's a better scope than "performance" for the commit title.

File: src/sentry/conf/server.py [hunk 1]
```diff
@@ -845,6 +845,8 @@ def create_partitioned_queues(name):
     "organizations:integrations-event-hooks": False,
     # Enable data forwarding functionality for organizations.
     "organizations:data-forwarding": True,
+    # Enable experimental performance improvements.
+    "organizations:enterprise-perf": False,
```
Comment (EvanPurkhiser) [inline src/sentry/conf/server.py:849]: Are we adding as a feature flag because we suspect we may cause regressions with attempts at making things faster?

**Review by evanpurkhiser — APPROVED** [review]
lgtm

**Review by leedongwei — APPROVED** [review]

File: src/sentry/conf/server.py (hunk 1 above)
Comment (slohmes) [reply src/sentry/conf/server.py:849]: That's a side benefit, the main goal is to create A/B segments to measure how much our changes improve performance relative to a control group.

**Pull request merged**